src/arrayMethodSort.js

```diff
@@ -4,8 +4,22 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compareFunction) {
+          if (compareFunction(this[j], this[j + 1]) > 0) {
+            [this[j], this[j + 1]] = [this[j + 1], this[j]];
+          }
+        } else {
+          if (String(this[j]) > String(this[j + 1])) {
+            [this[j], this[j + 1]] = [this[j + 1], this[j]];
+          }
+        }
```

Comment:
It's good that you're checking if a compareFunction is provided. However, you can simplify the logic by providing a default compareFunction when one is not provided. This way, you can avoid the if-else branching inside the loop and always use the compareFunction.